parser/src/token.rs

```diff
@@ -523,14 +545,19 @@ mod test {
 
     fn test(input: &str, expected: Vec<(&str, Token)>) {
         let mut tokenizer = tokenizer(input);
-
+        let mut count = 0;
+        let length = expected.len();
         for (token, (expected_span, expected_tok)) in tokenizer.by_ref().zip(expected.into_iter()) {
+            count = count + 1;
             println!("{:?}", token);
             let start = loc(expected_span.find("~").unwrap());
             let end = loc(expected_span.rfind("~").unwrap() + 1);
             assert_eq!(Ok(pos::spanned2(start, end, expected_tok)), token);
         }
 
+        assert_eq!(count, length);
+        assert_eq!(true, count > 0);
```

Comment:
👍  Never noticed it failed to check that expected did not contain additional tokens